features/nvstore/source/nvstore.h

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NVSTORE_H
+#define __NVSTORE_H
+
+// These addresses need to be configured according to board (in mbed_lib.json)
+#if defined(NVSTORE_AREA_1_ADDRESS) && defined(NVSTORE_AREA_1_SIZE) && \
+    defined(NVSTORE_AREA_2_ADDRESS) && defined(NVSTORE_AREA_2_SIZE) && \
+    defined(DEVICE_FLASH)
+#define NVSTORE_ENABLED 1
```

Comment:
what is the intention here with enabled flag - why cant this always be present?

Reply:
This was actually proposed by the folks from the Austin team. Right now, NVStore is currently supported on two boards (K64F & K82F) only. The defines configuring NVStore addresses and sizes must be there for each supported board, otherwise the module will not build properly. However, we don't want to fail the build for all other boards. So, any board that has these defines is considered to support NVStore, with the enabled define denoting it. All other boards will fail the build only when trying to access the NVStore class. In addition, the NVStore test is declared unsupported based on this define (much cleaner than checking all other defines again).